docs/GettingStarted.md

```diff
@@ -1617,3 +1620,57 @@ However, additional instrumentation provided by Datadog can be activated alongsi
 | `OpenTracing::FORMAT_BINARY`   | No         |                        |
+
+### Debugging
+
+If you are having difficulty using the tracer, the library offers a few debugging features to give you more insight into the internals of tracing.
+
+#### Using debug logging
+
+Debug logging can be activate to produce verbose logs that detail the operations of the tracer, by enabling it on the tracer settings:
+
+```ruby
+# config/initializers/datadog.rb
+require 'ddtrace'
+
+Datadog.configure do |c|
+  # Activates debug mode for the tracer
+  c.tracer debug: true
+end
+```
+
+Both the tracer and its integrations will produce detailed messages of operations to your configured log file.
+
+Be aware that *debug logging can produce many messages*, especially in applications with higher throughput. We generally recommend that this flag only be used in controlled, fail-safe environments (e.g. staging)
+
+#### Using debug stats
+
+To gather metrics regarding the internals of the trace library (such as number of traces flushed, number of errors, etc), one can enable debug metrics, which are collected using the `dogstatsd-ruby` library.
+
+To activate, first add `gem 'dogstatsd-ruby'` to your Gemfile, then add the following to your configuration file:
+
+```ruby
+# config/initializers/datadog.rb
+require 'datadog/statsd'
+require 'ddtrace'
+
+Datadog.configure do |c|
+  # Activates Statsd metric collection for debugging purposes.
+  # Configure with host and port of Datadog agent; defaults to 'localhost:8125'.
+  c.tracer statsd: Datadog::Statsd.new
+end
+```
+
+See the [Dogstatsd documentation](https://www.rubydoc.info/github/DataDog/dogstatsd-ruby/master/frames) for more details about configuring `Datadog::Statsd`.
+
+After activation, the tracer will send the following statistics:
+
+| Name                                               | Type    | Description                                                      |
+| -------------------------------------------------- | ------- | ---------------------------------------------------------------- |
+| `datadog.tracer.transport.http.client_error`       | `count` | Number of HTTP requests to agent with a client error.            |
+| `datadog.tracer.transport.http.incompatible_error` | `count` | Number of HTTP requests to agent with incompatible API.          |
+| `datadog.tracer.transport.http.internal_error`     | `count` | Number of internal tracer errors produced during HTTP transport. |
+| `datadog.tracer.transport.http.server_error`       | `count` | Number of HTTP requests to agent with a server error.            |
+| `datadog.tracer.transport.http.success`            | `count` | Number of successful HTTP requests to agent.                     |
+| `datadog.tracer.writer.services_flushed`           | `count` | Number of services flushed.                                      |
```

Comment:
You may stay more generic with this metric and use something like `datadog.tracer.services_flushed`